src/mdns/mdns_avahi.cpp

```diff
@@ -808,6 +820,7 @@ void PublisherAvahi::SubscribeService(const std::string &aType, const std::strin
 {
     auto service = MakeUnique<ServiceSubscription>(*this, aType, aInstanceName);
 
+    VerifyOrExit(mState == Publisher::State::kReady);
```

Comment:
add a log when SubscribeService failed ?

Reply:
Will do in a later PR